### DISCUSSION TITLE
user-agent rotation

### DISCUSSION THREAD

**Discussion by Keenpachi** [matin/garth#38] [Ideas]
Is there a option to add user agent rotation to increase number of requests before receiving "429 to many requests" error?
Or get a response with a number of request limits.

**Comment by matin** [comment]
How many requests and at what rate are you performing to get a 429?

**Reply by Keenpachi** [reply]
around 30 in 1 hour

**Reply by matin** [reply]
Are you using Garth directly or through a library that uses Garth?

I'm not able to reproduce this. I regularly perform 100s of requests in a few minutes.

**Reply by Keenpachi** [reply]
I'm using garminconnect -> https://github.com/cyberjunky/python-garminconnect/tree/master

I will try to test this using this standalone library

**Reply by matin** [reply]
garminconnect is fine. I was the one that integrated Garth into it, so it's done properly.

What information are you trying to export?

I just want to be able to reproduce the issue.

**Reply by Keenpachi** [reply]
I get the issue after running this over 30 times in 1h

`self.garmin = garminconnect.Garmin(self.email, self.password)`
`self.garmin.login()`
`all_records = self.garmin.get_activities(start=0, limit=100)`

**Reply by Keenpachi** [reply]
also resuming session from `~/.garth` is not working ->

`json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`
